pkg/controller/external_nofork.go

```diff
@@ -602,10 +601,10 @@ func (n *noForkExternal) Create(ctx context.Context, mg xpresource.Managed) (man
 	}
 	n.opTracker.SetTfState(newState)
 
-	if _, err := n.setExternalName(mg, newState); err != nil {
+	stateValueMap, err := n.fromInstanceStateToJSONMap(newState)
+	if _, err := n.setExternalName(mg, stateValueMap); err != nil {
```

Comment:
Let's make sure we check any errors returned from `n.fromInstanceStateToJSONMap`:
```suggestion
	stateValueMap, err := n.fromInstanceStateToJSONMap(newState)
	if err != nil {
		return managed.ExternalCreation{}, errors.Wrap(err, "failed to create the resource")
	}
	if _, err := n.setExternalName(mg, stateValueMap); err != nil {
```

Reply:
Done!